### PR TITLE
Water Rendering

### PR DESCRIPTION
**Pull request by VectorIV**
Added simple water rendering overview page including figure images.

### CODE REVIEW

**Comment by feliwir** [comment]
@VectorIV could you export the Photoshop files as PNG and include them, since JPG tends to be a bit blurry for lineart. Also please remove the original PSD files if possible. Apart from that it looks great :+1:

File: .vs/VSWorkspaceState.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (feliwir) [inline .vs/VSWorkspaceState.json]: Please don't commit your ".vs" folder

File: posts/water_rendering/water_rendering.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Refraction is fairly simple to achieve. You only need to render it the parts that's below the water plane.
+So in a nutshell:
+1. Using the current player camera and clipping plane, render to a new frame buffer.
+2. Use the frame buffer as a texture on the plane and apply shader techniques to it to make it looks convincing.
+
+![Refraction Rendering](./2.png)
+*<center>Figure 2: Camera For Refraction Rendering</center>*
+
+## Waves
+This is done at shader level either via vertex or fragment. Many techniques could be used here including DuDv mapping, bump mapping, nomral mapping, etc.
```
Comment (Skleni) [inline posts/water_rendering/water_rendering.md]: Typo: nomral

File: posts/water_rendering/water_rendering.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+We'll using only fragment for this implementation.
+
+**DuDv Mapping** is simply a directoinal distortion map usually derived from a normal map. It only has R and G channels for offsetting U and V vaules of the texture. This creates ripple effect on the surface.
```
Comment (Skleni) [inline posts/water_rendering/water_rendering.md]: Typos: directoinal, vaules, effect(s?)

File: posts/water_rendering/water_rendering.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+![DuDv Map](./wdudv.jpg)
+*<center>Figure 3: Sample DvDu Map</center>*
+
+**Normal Mapping** is an extensive bump map uses for lighting calculation. Having all RGB channels, they are represented as XYZ directions. This creates the illusion of detailed surface without having a lot of polygons.
```
Comment (Skleni) [inline posts/water_rendering/water_rendering.md]: used insted of uses?
...of _a_ detailed surface?

File: posts/water_rendering/water_rendering.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+![Normal Map](./wnormal.jpg)
+*<center>Figure 4: Sample Normal Map</center>*
+
+## Caustics
+This is also done at shader level as well using fragment. The simplest and the only way I know how to do is using animated textures. By having a set of pre-baked textures, one can animate it to simulate the effects of caustics.
+At higher level of shadering, I assume one may simulate casutics from the movement of the waves and direction of the lightsource.
```
Comment (Skleni) [inline posts/water_rendering/water_rendering.md]: Typo: casutics

File: posts/water_rendering/water_rendering.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+In a nutshell, you use add blending to apply the animated textures to terrain and objects under the water plane with respect to the depth of said subjects. The deeper the subject, the more transparent the apply should be.
+![Caustics Map](./cause.png)
+*<center>Figure 5: Samples of Caustics Map</center>*
+
+![Caustics Rendering](./3.png)
+*<center>Figure 6: Caustics Rendering</center>*
+
+## Miscellaneous
+**Fresnel Effect** A principle of reflectivity. Shallow angle = strong reflectivity, steep nagle weak reflectivity. By applying this principle to reflection and refraction rendering, one can improve the realism of water rendering.
```
Comment (Skleni) [inline posts/water_rendering/water_rendering.md]: Typo: nagle
missing = before weak?

File: posts/water_rendering/water_rendering.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+![Fresnel Rendering](./4.png)
+*<center>Figure 6: Fresnel Rendering</center>*
+
+## Result
+Combining those components together, you get the result down below: an improved version of vanilla water from early reveal screenshots. While not the fanciest water you might have ever seen but still, it looks more watery than AoE4's attempt.
+![Result](./result.png)
+With time and afford, you can make it even more convincing but that is a discussion for another time.
```
Comment (Skleni) [inline posts/water_rendering/water_rendering.md]: effort?

**Review by Skleni — CHANGES_REQUESTED** [review]
Found some typos...otherwise, great work! :)

**Comment by Skleni** [comment]
Why are the roads so broken in the result image? Is this based on the current codebase?

**Review by Skleni — APPROVED** [review]
Thanks!

**Pull request merged**